web/pdf_page_detail_view.js

```diff
@@ -0,0 +1,273 @@
+      // visible area, but we ignore the .maxCanvasPixels limit.
+      // TODO: We should probably instead give up and not render the detail view
+      // in this case. It's quite rare to hit it though, because usually
+      // .maxCanvasPixels will at least have enough pixels to cover the visible
+      // screen.
+    }
+
+    const overflowWidth = visibleWidth * overflowScale;
+    const overflowHeight = visibleHeight * overflowScale;
+
+    const minX = Math.max(0, visibleArea.minX - overflowWidth);
+    const maxX = Math.min(viewport.width, visibleArea.maxX + overflowWidth);
+    const minY = Math.max(0, visibleArea.minY - overflowHeight);
+    const maxY = Math.min(viewport.height, visibleArea.maxY + overflowHeight);
+    const width = maxX - minX;
+    const height = maxY - minY;
+
+    this.#detailArea = { minX, minY, width, height, scale: viewport.scale };
+
+    this.reset({ keepCanvas: true });
+  }
+
+  async draw() {
+    // The PDFPageView might have already dropped this PDFPageDetailView. In
+    // that case, simply do nothing.
+    if (this.pageView.detailView !== this) {
+      return undefined;
+    }
+
+    // If there is already the lower resolution canvas behind,
+    // we don't show the new one until when it's fully ready.
+    const hideUntilComplete =
+      this.pageView.renderingState === RenderingStates.FINISHED ||
+      this.renderingState === RenderingStates.FINISHED;
+
+    if (this.renderingState !== RenderingStates.INITIAL) {
+      console.error("Must be in new state before drawing");
+      this.reset(); // Ensure that we reset all state to prevent issues.
+    }
+    const { div, pdfPage, viewport } = this.pageView;
+
+    if (!pdfPage) {
+      this.renderingState = RenderingStates.FINISHED;
+      throw new Error("pdfPage is not loaded");
+    }
+
+    this.renderingState = RenderingStates.RUNNING;
+
+    const canvasWrapper = this.pageView._ensureCanvasWrapper();
+
+    const { canvas, prevCanvas, ctx } = this._createCanvas(newCanvas => {
+      // If there is already the background canvas, inject this new canvas
+      // after it. We cannot simply use .append because all canvases must
+      // be before the SVG elements used for drawings.
+      if (canvasWrapper.firstElementChild?.tagName === "CANVAS") {
+        canvasWrapper.firstElementChild.after(newCanvas);
+      } else {
+        canvasWrapper.prepend(newCanvas);
+      }
+    }, hideUntilComplete);
+    canvas.setAttribute("aria-hidden", "true");
+
+    const { width, height } = viewport;
+
+    const area = this.#detailArea;
+
+    const { devicePixelRatio = 1 } = window;
+    const transform = [
+      devicePixelRatio,
+      0,
+      0,
+      devicePixelRatio,
+      -area.minX * devicePixelRatio,
+      -area.minY * devicePixelRatio,
+    ];
+
+    canvas.width = area.width * devicePixelRatio;
+    canvas.height = area.height * devicePixelRatio;
+    canvas.style.position = "absolute";
```

Comment:
nit: maybe it can be moved in the css.

Reply:
Oh actually it's already defined in the CSS, I can just delete these two lines.